pkg/endpoint/bpf.go

```diff
@@ -206,8 +206,14 @@ func (e *Endpoint) writeHeaderfile(prefix string, owner Owner) error {
 	fw.WriteString(" */\n\n")
 
 	// If policy has not been derived or calculated yet, all packets must
-	// be dropped until the policy of the endpoint has been determined.
-	if !e.PolicyCalculated && owner.PolicyEnforcement() != NeverEnforce {
+	// be dropped until the policy of the endpoint has been determined,
+	// except when it is known that the current policy will not drop anything,
+	// which is true when:
+	// - policy enforcement mode is "never"
+	// - policy enforcement mode is "default" and no policies are loaded
+	if !e.PolicyCalculated &&
+		!(owner.PolicyEnforcement() == NeverEnforce) &&
+		!(owner.PolicyEnforcement() == DefaultEnforcement && owner.GetPolicyRepository().Empty()) {
```

Comment:
I might be very wrong, but can't we also control policy enforcement per-endpoint? (so both checks above would have to be a check on the endpoint's policy object, or the policy of it's identity). I'm basing this on some docs I read, not on reading the code (http://cilium.readthedocs.io/en/latest/policy/intro/#policy-enforcement-modes), and they seem to suggest that policyEnforcement=default acts per-endpoint.

Reply:
Oh, actually, it might need to check the global and local setting. aanm has suggested an edge-trigger to propagate global setting changes to the endpoint setting but we don't currently implement that. This might mean that in some combinations the global value trumps the local one (global: enforceAlways and local:default).